gems/pending/util/esx_thumb_print.rb

```diff
@@ -0,0 +1,34 @@
+require 'util/thumb_print'
+
+class ESXThumbPrint < ThumbPrint
+  attr_reader :user, :password, :cert
+
+  def initialize(host, user, password)
+    @user     = user
+    @password = password
+    super(host)
+  end
+
+  def uri
+    url  = "https://" + @host + "/host/ssl_cert"
+    @uri = URI(url)
+    @uri
+  end
+
+  def http_request
+    super
+    @request          = Net::HTTP::Get.new(@uri.request_uri)
```

Comment:
changing to call uri.request_uri. but the instance variable is still needed because ThumbPrint.http_request needs it.  And note that ThumbPrint.new will always call uri before http_request so unless someone removes the call to uri @uri can't be nil.